src/objects.rs

```diff
@@ -1277,9 +1282,81 @@ pub struct ODRule {
     pub network_id: Option<String>,
     pub physical_mode_id: Option<String>,
 }
+impl_id!(ODRuleV1);
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[cfg(feature = "fare-v2")]
+pub struct Ticket {
+    pub id: String,
+    pub name: String,
+    pub comment: String,
+}
+#[cfg(feature = "fare-v2")]
 impl_id!(Ticket);
-impl_id!(ODRule);
+
+#[cfg(feature = "fare-v2")]
+impl GetObjectType for Ticket {
+    fn get_object_type() -> ObjectType {
+        ObjectType::Ticket
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[cfg(feature = "fare-v2")]
+pub struct TicketPrice {
+    pub ticket_id: String,
+    pub price: f64, // TODO: how to ensure this value is always positive (u64 on cents? Setter?)
```

Comment:
Any idea how to deal with this in the model? One solution would be to have an Integer, so we would give the total number of cents. I'm not a fan of this solution.  The other solution would be to have either a setter on this field or a constructor on this `struct`, but that doesn't prevent us from doing
```
TicketPrice {
	ticket_id: "some_id",
	price: -42,
	...
}
```

Reply:
you could do a custom deserialization and check in it if the value is negative

Reply:
Thanks for the nice suggestion... and then I did in 9892efcff7f7e2cf9cd80f7660a4357e65b19ff8.